Mixxx 2.6 beta release (#389)
mixxxdj/website: content/news/XXXX-XX-XX-mixxx-2_6_beta-released.md

We're proud to announce a new beta release of Mixxx, version 2.6.
This is a beta release containing a lot of new features, we offer this release in order to get it tested by the most possible users.
We look forward to the feedback of all users.

Mixxx needs testers for the new 2.6 beta release.
If you would like to contribute to your favorite DJ program (translate, add a mapping, add a feature, test) [Get Involved](https://mixxx.org/get-involved/).

Enjoy Mixxx 2.6 beta

#### STEMS STEMS STEMS

To explain what STEMS are all about, we take a sentence from the introduction of this post, a regular text and convert it to a STEMS-text.
In this example we extract 4 TEXT-STEMS that represent each a particular part of that sentence that only result in the original text when they are combined they result.

In the Text stems example here you can play with the mute and effect buttons.
With the MUTE button you can make the Channel disappear, with the Effect button you can add an effect to the Channel.
To see the original text each Channel needs to be un-muted and without effect.

<div class="stemdemo">
  <style>
    body {
      font-family: Raleway;
      padding: 20px;

>     }

Reviewer: this is applied to the entire page (`body`), not just the script part.
**With** that, it's now using some serif for entire page.
When I deactivate it, all looks good :shrug:
I don't have the railway font, is it downloaded by the page? (it's been very long since I did (basic) webdev..)

the broken view:
![image](https://github.com/user-attachments/assets/e2a98912-b4cf-49d8-a395-eb5e7fdb6536)

Author: as Daniel mentioned earlier raleway is the website font, when there was an other font mentioned in the css of the script it took over the whole page. This way it looks like it's meant to be.
So which font do you see if you don't have raleway?

Author: I removed the body { font-family: Raleway; padding: 20px;  }
I inspected on different browsers / machines, it looks ok
@ronso0  & @daschuer can you please check?

status: draft

#### Mixxx 2.6 beta Release Announcement

We're proud to announce a new beta release of Mixxx, version 2.6.
This is a beta release containing a lot of new features, we offer this release in order to get it tested by the most possible users.
We look forward to the feedback of all users.

Mixxx needs testers for the new 2.6 beta release.
If you would like to contribute to your favorite DJ program (translate, add a mapping, add a feature, test) [Get Involved](https://mixxx.org/get-involved/).

Enjoy Mixxx 2.6 beta

#### STEMS STEMS STEMS

To explain what STEMS are all about, we take a sentence from the introduction of this post, a regular text and convert it to a STEMS-text.
In this example we extract 4 TEXT-STEMS that represent each a particular part of that sentence that only result in the original text when they are combined they result.

In the Text stems example here you can play with the mute and effect buttons.
With the MUTE button you can make the Channel disappear, with the Effect button you can add an effect to the Channel.
To see the original text each Channel needs to be un-muted and without effect.

<div class="stemdemo">
  <style>
    body {

>       font-family: sans-serif;

Reviewer 2: This font is applied to the whole article. Deleting this line fixes the issue. But I guess we have other issue with this embedded block.

Author: I'll change the font in the script and adapt the hash

Author: this will be better

Author: is it OK now?

        }
      )
    );

    canvas.innerHTML = '';
    for (let i = 0; i < baseText.length; i++) {
      const char = baseText[i];
      const channel = getCharChannel(char);

      canvas.innerHTML += `<span class="c${channel}">${char}</span>`;
    }
  </script>
</div>

Wouldn't it be nice if we could do this tricks with music?
This possibility exists and is called STEMS. Stems are audiotracks that contain 4 stems-tracks: Drums, Bass, Melody (or other) and Vocals (or Vox).

#### STEMS in Mixxx

With Mixxx starting with this 2.6 beta release you can control the Volume and Effects for each of these Stem-tracks independently.
At the moment you need to create the STEM-Tracks with 3rd party tools, but once added to the Mixxx-library you can control them like
- create an a cappela: Play Drums, Bass and Other "Normal" + mute the Vocals
- create a drum pattern: Play Drums normally, mute Bass, Other and Vocals
- remix a track: set the volume of the Drums on 100%, the bass on 90%, the other on 50% and the vocals on 60%
- add effects to the vocal only: set the volume of all stems to a value of your choice + add reverb effect only to the vocals.

> Mixxx wouldn't be Mixxx if these controls weren't available for your controller, so they arthis.

Reviewer: ```suggestion
Mixxx wouldn't be Mixxx if these controls weren't available for your controller, so they are.
```

To control the Stems you need for each channel, depending on your choice
- 4 buttons if you want to use the 'mute' functions
- 4 encoders to control the Volume
- 4 buttons to activate the effect
- 4 encoders to control the effect level.

Find more info about Stems in Mixxx in [Antoine Colombier's GSoC Article](https://mixxx.org/news/2024-08-26-stem-mixing/)

@Video(https://www.youtube.com/watch?v=oEJuvVFSbnE)

#### CUES CUES CUES

> What if you wanted to add an extra cue, that logically should be between the 2nd and 3rd cue you created? In 2.6 beta this problem is solved. You can move a cue to another pôsition without the need to re-create the cue. It's also possible to rearrange all cues according to their position in the track.

Reviewer: ```suggestion
What if you wanted to add an extra cue, that logically should be between the 2nd and 3rd cue you created? In 2.6 beta this problem is solved. You can move a cue to another position without the need to re-create the cue. It's also possible to rearrange all cues according to their position in the track.
```

![Drop a hotcue on another position]({static}/images/news/Mixxx_2_6_beta_drop_hotcue_on_other_position.gif)

> You can also use a menu action to re-order the hot cues on their position in the track.

Reviewer: ```suggestion
You can also use a menu action to re-order the hot cues according to their position in the track.
```

![Drop a hotcue on another position]({static}/images/news/Mixxx_2_6_beta_menu_reorder_hotcues.gif)

> Before cues played as long as you pressed them, to continue the playing you needed to press the play button. That's solved too: press the cue, drag & drop it on the play button and ... tada the track keeps playing.

Reviewer: ```suggestion
Before, cues played as long as you pressed them, to continue the playing you needed to press the play button. That's now possible with only a mouse: click and hold the cue, drag & drop it on the play button and ... tada, the track keeps playing.
```

title: Mixxx 2.6 beta Released
authors: Evelynne Veys
tags: 2.6 beta, release announcement start  CI
comments: yes
status: draft

#### Mixxx 2.6 beta Release Announcement

We're proud to announce a new beta release of Mixxx, version 2.6.
This is a beta release containing a lot of new features, we offer this release in order to get it tested by the most possible users.
We look forward to the feedback of all users.

The major subject in this release is STEMS.

#### STEMS STEMS STEMS

> To explain what STEMS are all about, we take the introduction of this post, a regular text and convert it to a STEMS-text.

Reviewer: Does it really need to be that complicated?
I mean, we're talking to musicians here, so they'll get the usual 'instruments separation' explanation, no?

Author: We're also talking to people without any musical skill and beginners.
I stopped counting the times I needed to explain what Stems are, so I 'd like to explain it BASIC.

Beginners Appreciate Simple Instructions for Complexity (Sorry changed that a bit 😄)

Reviewer: Well, how did yo do it? ; )

https://www.marmosetmusic.com/journal/what-are-song-stems-and-why-do-you-need-them/
> Song stems are individual musical elements that, when layered together, make a song.

https://www.native-instruments.com/en/specials/stems/
> A Stem file is a multi-channel audio file that contains a track split into four musical elements – bass, drums, vocals, and melody for example.

These are one sentence each. And if that doesn't suffice for some users, there's youtube.

Here I see a wall of text.. TL;DR :shrug:

Author: You don't like the idea or do you find it's too long (or both)?

Would you prefer it without images:

Mixxx needs testers for the new 2.6 beta release.
M_xxx n__ds t_st_rs f_r th_ n_w 2.6 b_t_ r_l__s_.
_____ _____ _______ ___ ___ ___ ___ ___a ____a___
_____ _ee__ _e__e__ ___ __e _e_ ___ _e__ _e_e__e_
_i___ _____ ______ _o_ ___ ___ ___ ____ ________
_i___ _ee__ _e__e__ _o_ __e _e_ ___ _e_a _e_ea_e_
Mixxx neeeeds teesteers for the neew 2.6 beeta reeleeasee.

Reviewer 3: I like the idea! I think current look maybe a little bit too confusing tho. 
I'm wondering if we could do this with some JS instead and have it more interactive? Perhaps this is "simply enough" for an AI to generate for us? I'm mean we are at 70%of AGI so surely it canmdo that, right? :sweat_smile:

Reviewer: I like the _idea_, it's playful but not as clear as waveforms (either Mixxx waveforms or DAW/score-like).
And I'd use in a much simpler way so it extends the simple musical explanation.

With one sentence it could be comprehensible. First I thought it is a ridle..

Reviewer: Yeah :+1: for an interactive version

Reviewer: And we also need to keep in mind that the news are viewed on mobile, too.

#### STEMS in Mixxx

With Mixxx starting from with this 2.6 beta release you can control the Volume and Effects for each of these Stem-tracks independently.
At the moment you need to create the STEM-Tracks with 3rd party tools, but once added to the Mixxx-library you can control them like
- create an a cappela: Play Drums, Bass and Other "Normal" + mute the Vocals
- create a drum pattern: Play Drums normally, mute Bass, Other and Vocals
- remix a track: set the volume of the Drums on 100%, the bass on 90%, the other on 50% and the vocals on 60%
- add effects to the vocal only: set the volume of all stems to a value of your choice + add reverb effect only to the vocals.

Mixxx wouldn't be Mixxx if these controls weren't available for your controller, so they are.
To control the Stems you need for each channel, depending on your choice
- 4 buttons if you want to use the 'mute' functions
- 4 encoders to control the Volume
- 4 buttons to activate the effect
- 4 encoders to control the effect level.

Find more info about Stems in Mixxx in [Antoine Colombier's GSoC Article](https://mixxx.org/news/2024-08-26-stem-mixing/)

Enjoy Mixxx

#### 2.6 beta Changelog

> The complete changelog can be found [here](https://)

Reviewer 2: ```suggestion
The complete changelog can be found [here](https://github.com/mixxxdj/mixxx/blob/2.6-beta/CHANGELOG.md)
```
